src/msdo-nuget-client.ts

```diff
@@ -229,50 +229,137 @@ async function _resolveVersion(
     let result = await requestJson(searchQueryServiceUrlWithQuery, requestOptions);
     const findPreRelease = common.isLatestPreRelease(packageVersion);
 
+    resolvedVersion = findLatestVersion(result, findPreRelease);
+    
+    if (resolvedVersion == null) {
+        throw new Error(`Package not found: ${packageName}`);
+    }
+
+    return resolvedVersion;
+}
+
+/**
+ * Finds the latest version in a SearchQueryService response.
+ * 
+ * @param result A JSON object returned from the SearchQueryService
+ * @param findPreRelease Whether or not to find a pre-release version
+ * @returns The latest version of the package
+ */
+function findLatestVersion(
```

Comment:
>findLatestVersion [](http://example.com/codeflow?start=9&length=17)

Don't have to address it in this PR, but this logic seems like a good candidate where unit tests would be helpful